### DISCUSSION TITLE
How to have a custom list field where user can enter free text?

### DISCUSSION THREAD

**Discussion by fred-the-coder** [joomengine/Joomla-Component-Builder#646] [Q&A] [answered]
Hello,
I have a custom JCB field, retrieving the list of items from the database.

But actually, I need to be able to let the user enterring a value manually when the ones already listed in the dropdow list do not have the expected value.

How to do that?
Seems like "combo" joomla form field does this (https://docs.joomla.org/Combo_form_field_type)...but it is not available in JCB...

Thank you for your help.

N.B.: I tried to update my custom filed, by setting "true" to button property.
But the add new button is not yet implemented in frontend edit view...

**Comment by Llewellynvdm** [comment, marked as answer]
JCB Field_types are build in a way, that the whole idea is that you can add field_types as you need them, and only if you've added them and find problems... will we come in and help adjust JCB to adapt to the new field_type.

I looked and combo is already mentioned in the field [checker area](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/helpers/componentbuilder.php#L3667).  So I added it, and update the field_type_updater so you can pull this field into an existing JCB install.

Then testing it I found it is not working as expected.... so I am debugging it and will get back to you.

**Reply by Llewellynvdm** [reply]
Okay I found the issue, and now the combo field works.

Here is a [new tutorial](https://youtu.be/kbn-DF6zXlo) that will get you going... :+1:

**Comment by fred-the-coder** [comment]
@Llewellynvdm thank you very much.
As you found some bugs (_Then testing it I found it is not working as expected...._) does it mean, I have to update JCB or at least retrieve your latest fixes?
Thank you

**Comment by fred-the-coder** [comment]
Any clue on how to fill the combo choices with data from a Joomla table?
As you explained for custom Fields?
Maybe I have to open a new forum item...? please, let me know.

**Reply by Llewellynvdm** [reply]
The same way you can extend any existing field with custom-custom fields, watch these tutorials:

[Extended explanation to JCB Custom Field](https://www.youtube.com/watch?v=91iIAuHZj38&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=73)
[Extend Joomla fields via JCB](https://www.youtube.com/watch?v=Mu9H0zgH9Lw&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=75)

You will need to know how to expand the compo field type to achieve your objective, but it is very doable.

**Reply by fred-the-coder** [reply]
I did it!

Thank you again for your support.
